draft-ZIP-0227.rst

```diff
@@ -0,0 +1,184 @@
+in ZIP 227 [#zip-0227]_. The transfer and burn of such assets is defined in ZIP 226 [#zip-0226]_.
+
+Motivation
+==========
+
+The current Orchard protocol does not support custom assets, which is an important aspect for the proper
+evolution of the Zcash ecosystem. Enabling multi-asset support on the Zcash chain will open the door for a host of
+applications and enhance the ecosystem with application developers and asset custody institutions for issuance and
+bridging purposes.
+
+Overview & Rationale
+====================
+
+This ZIP specifies the issuance mechanism of the Zcash Shielded Assets (ZSA) protocol and must work conjunction with ZIP 226 [#zip-0226]_, as the issuance mechanism is only valid for the ZSA transfer protocol, as it produces notes that can only be transferred under ZSA.
+
+In short, we are going to enable only *transparent* issuance, since we believe that as a first step, transparency is important to properly test the applications that will be most used in the Zcash ecosystem, while enabling tracking of supply of assets. In essence, any user of the blockchain can issue new asset types, yet only a single issuer (a set of keys) can issue that specific asset type, as the asset identifiers are tied to the issuer keys.
+
+We have designed the protocol such that at the time of issuance, the issuer can already allocate all the tokens to the corresponding owners by creating the corresponding (shielded) output notes to the corresponding addresses. As is implied, the issuance mechanism is itself transparent, but the allocation is totally private. Furthermore, in a single issuance transaction, the issuer can create multiple asset types (and allocate all of those too). Finally, a key component of our design is the `finiteIssuance` boolean that defines whether a specific asset type can have more tokens issued or not.
+
+The issuance mechanism is broad enough for issuers to either create native assets on Zcash (i.e.: assets that originate on the Zcash blockchain), as well as for institutions to create bridges from other chains and import wrapped assets. In fact, the protocol described below enables, what we hope will be a useful set of, applications:
+
+- By setting the `finiteIssuance = 1` from the first issuance instance of that asset type, the issuer is in essence creating a one-time issuance transaction. This is useful when the max supply is capped from the beginning and the distribution is known in advance. All tokens are issued at once and distributed as needed.
+- When the `finiteIssuance = 0`, the issuer can keep creating tokens of that type (in a transparent manner). The boolean value can be changed with any new issuance transaction, and could be set to `1` either when the issuer keys have been compromised, and hence stopping all issuances of that asset type (the boolean cannot be reversed), or whenever the issuer decides that the max supply has been reached.
```

Comment:
```suggestion
- When the `finiteIssuance = 0`, the issuer can keep creating tokens of that type (in a transparent manner). The boolean value can be set to `1` with any new issuance transaction, either when the issuer keys have been compromised in order to stop all issuances of that asset type (since the boolean cannot be reversed), or whenever the issuer decides that the maximum supply has been reached.
```

Reply:
Do we want to replace "reversed" with "set to 0 once it has been set to 1"?

Reply:
[PR#649] has rewritten this portion significantly, so the suggestions have not been directly applied, but the rewritten file addresses the comments.